nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/utils/AzureStorageUtils.java

```diff
@@ -311,4 +316,38 @@ public static void setProxy(final OperationContext operationContext, final Proce
         final ProxyConfiguration proxyConfig = ProxyConfiguration.getConfiguration(processContext);
         operationContext.setProxy(proxyConfig.createProxy());
     }
+
+    public static void configureProxy(final NettyAsyncHttpClientBuilder nettyClientBuilder, final PropertyContext propertyContext) {
+        final ProxyConfiguration proxyConfiguration = ProxyConfiguration.getConfiguration(propertyContext);
+
+        if (proxyConfiguration != ProxyConfiguration.DIRECT_CONFIGURATION) {
+
+            final ProxyOptions proxyOptions = new ProxyOptions(
+                    getProxyType(proxyConfiguration),
+                    new InetSocketAddress(proxyConfiguration.getProxyServerHost(), proxyConfiguration.getProxyServerPort()));
+
+            final String proxyUserName = proxyConfiguration.getProxyUserName();
+            final String proxyUserPassword = proxyConfiguration.getProxyUserPassword();
+            if (proxyUserName != null && proxyUserPassword != null) {
+                proxyOptions.setCredentials(proxyUserName, proxyUserPassword);
+            }
+
+            nettyClientBuilder.proxy(proxyOptions);
+        }
+    }
+
+    private static ProxyOptions.Type getProxyType(ProxyConfiguration proxyConfiguration) {
+        if (proxyConfiguration.getProxyType() == Proxy.Type.HTTP) {
+            return ProxyOptions.Type.HTTP;
+        } else if (proxyConfiguration.getProxyType() == Proxy.Type.SOCKS) {
+            final SocksVersion socksVersion = proxyConfiguration.getSocksVersion();
+            if (socksVersion != SocksVersion.NOT_SPECIFIED) {
+                return ProxyOptions.Type.valueOf(socksVersion.name());
+            } else {
+                throw new IllegalArgumentException("Version should be set when SOCKS proxy type is used!");
```

Comment:
Use of exclamations in error messages should be avoided. Recommend adjusting the wording for greater clarity:
```suggestion
                throw new IllegalArgumentException("SOCKS Version is required for SOCKS Proxy Type");
```